### ISSUE TITLE
options --enable-transparent-visuals --disable-gpu seem to be unnecessary . Deprecate it ?

### ISSUE THREAD

**Issue by doom-fr** [electron/electron#16808]
* **Electron Version** (output of `node_modules/.bin/electron --version`):
  * v4.0.1
* **Operating System** (Platform and Version):
  * debian jessie
* **Last known working Electron version** (if applicable):
  * ?

### Expected Behavior
transparency doesn't works without these options

### Actual behavior
works with or without these options

### Additional Information
Is these options still required ?
The [Nvidia Alpha issue](https://bugs.chromium.org/p/chromium/issues/detail?id=369209) has been fixed in 2016.

Shouldn't we update this in https://electronjs.org/docs/api/frameless-window
And maybe deprecate `--enable-transparent-visuals` option ?

**Comment by clytras** [comment]
@doom-fr I can't get transparency to work on Ubuntu 19.10 VM without `--enable-transparent-visuals`, even when using a delay >= 300 ms to create the window after app ready.

Electron version **7.1.2**

**Comment by doom-fr** [comment]
Hi @clytras and does it work for you with the option ?

**Comment by clytras** [comment]
@doom-fr I am very confused! I got it working with just only `--enable-transparent-visuals` and without a delay, but just once! Then, after VM restart, whatever I did it didn't work! Now it only works if I create the window with a delay >= 400 ms, with or without `--enable-transparent-visuals`! Really strange!

**Comment by doom-fr** [comment]
Here is a stackoverflow answer that can help you : https://stackoverflow.com/a/53612021/1619521

But maybe you have already find it.

**Comment by clytras** [comment]
@doom-fr of course I've found it, I have even upvoted it. That is how it's working for me now. But the strange thing is that I got it working without a delay and with `--enable-transparent-visuals` just once! I'll try some more tests on a clean install.

**Comment by clytras** [comment]
I figured out what happened! I've had `app.on('ready', async ...` event with an async function to fetch `app.getGPUInfo` promise result. This promise takes some time to resolve, for sure more than 500 ms on development. My previous test case had `BrowserWindow` transparent to false and when I tried `--enable-transparent-visuals`, I immediately set transparent to true, so the transparency worked because of the `app.getGPUInfo` promise delay! Can't believe it!

**Comment by doom-fr** [comment]
For information, I proposed this https://github.com/electron/electron/issues/16809 to solve the problem.

**Comment by clytras** [comment]
Yes, I've read this also. It is a bug and needs to be fixed. It's like the render system isn't ready when the ready event is triggered.

**Comment by fabiospampinato** [comment]
> Is these options still required ?

My app doesn't render inside VMs without `--disable-gpu`, so I wouldn't remove it.

**Comment by felixrieseberg** [comment]
We have overhauled our documentation in recent years, so this issue is no longer relevant in its current form. If any of you disagree, please open a new one!
